core/arch/arm/tee/entry_std.c

```diff
@@ -110,6 +111,14 @@ static TEE_Result set_rmem_param(const struct optee_msg_param_rmem *rmem,
 	mem->offs = READ_ONCE(rmem->offs);
 	mem->size = READ_ONCE(rmem->size);
 
+	/*
+	 * Check that the supplied offset and size is covered by the
+	 * previously verified MOBJ.
+	 */
+	if (ADD_OVERFLOW(mem->offs, mem->size, &req_size) ||
+	    mem->mobj->size < req_size)
+		return TEE_ERROR_SECURITY;
```

Comment:
It's an **or** operation and not an **and** operation. This seems perfectly legit to me. I.e., if one of the two conditions is true, then this shall fail. For an **and** operation I think it would still write it like this, but there I'm more with you regarding the suggestion. Agree?